### PR TITLE
Use signing date when deriving signing key

### PR DESCRIPTION
**Pull request by dagnir**
## Description
Fix an issue where the signing key is created only once at the start of the
request for event streaming requests. This causes requests that span two more
more days to have signing errors once the date changes because the signing key
was derived only once using the date at the beginning of the request.

## Motivation and Context
Bug fix.

## Testing
 - `mvn clean install`
 - ~~Running integ tests~~ integ test

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/auth/src/test/java/software/amazon/awssdk/auth/signer/internal/SignerKeyTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+package software.amazon.awssdk.auth.signer.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import java.time.Instant;
+import org.junit.Test;
+
+public class SignerKeyTest {
+
+    @Test
+    public void isValidForDate_dayBefore() {
```
Comment (cenedhryn) [inline core/auth/src/test/java/software/amazon/awssdk/auth/signer/internal/SignerKeyTest.java]: Shouldn't it be isNotValid? and for the after method too

Comment (dagnir) [reply core/auth/src/test/java/software/amazon/awssdk/auth/signer/internal/SignerKeyTest.java]: I normally like to name the tests to include the name of the method I'm testing, which is `isValidForDate()` in this case :)

Comment (cenedhryn) [reply core/auth/src/test/java/software/amazon/awssdk/auth/signer/internal/SignerKeyTest.java]: I like to include the expected behavior to be sure I understand the test's intention.

Comment (dagnir) [reply core/auth/src/test/java/software/amazon/awssdk/auth/signer/internal/SignerKeyTest.java]: That's reasonable. Will update

**Pull request merged**